### ISSUE TITLE
Alternative transcripts, GO annotation

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#1012]
I also spotted this anomaly, might be related?

tcio4 has been explicitly annotated to "plasma membrane"
<img width="1014" alt="Screenshot 2022-08-02 at 20 28 20" src="https://user-images.githubusercontent.com/7359272/182544316-c3767476-e26e-4324-ba40-56d2c14e9a5e.png">

the other transcript (not annotated purposely) also gets this annotation from GOA.
So somehow the GOA annotation is converted to an isoform specific allotation

(note that the other component annotation is not correct,  but it is displayed as curated)

**Comment by ValWood** [comment]
Also not that I'm not sure tco1 should be curated as an alternative transcript. This was part of an ongoing discussion
https://github.com/pombase/pombase-chado/issues/865
For now, it is a useful test case.

**Comment by kimrutherford** [comment]
> So somehow the GOA annotation is converted to an isoform specific allotation

It looks different for me:
![image](https://user-images.githubusercontent.com/90474/183003976-e484b7ad-410d-48ef-9be1-b080c3ea1d02.png)

We you looking at the term page?

Either way, it's definitely wrong because GOA annotation should be applied to both transcripts.

**Comment by ValWood** [comment]
Gene page.

**Comment by kimrutherford** [comment]
Are you sure?  We don't show the gene name in the annotation table on the gene pages because it's redundant.

**Comment by ValWood** [comment]
We do if there are altrernative transcript annotations (click on 'show all' icon)

**Comment by ValWood** [comment]
<img width="1119" alt="Screenshot 2022-08-05 at 07 02 16" src="https://user-images.githubusercontent.com/7359272/183011827-778458a5-f1b1-4923-bc45-b66a3c7fa88a.png">

**Comment by ValWood** [comment]
Also if the annotation is to  th main gene ID, not to a transcript ID, I don't think we should display "All' we should probably leave it blank. If we know it is both, we should add both IDs. 

but most of the imported annotations  could be one, or the other or both so saying "All" would not always be correct.

**Comment by kimrutherford** [comment]
> We do if there are altrernative transcript annotations (click on 'show all' icon)

In your first screenshot there is a gene name column with "tco1" and no term columns.  Which is why I asked if that was from the term page.

**Comment by kimrutherford** [comment]
> Also if the annotation is to th main gene ID, not to a transcript ID, I don't think we should display "All'

Should we make that change on the term page too?: https://www.pombase.org/term/GO:0110134

The difference there is that there is no other way to tell if a gene has multiple transcripts.  We currently have a blank in the transcript ID column if the gene has only one transcript.

**Comment by kimrutherford** [comment]
> most of the imported annotations could be one, or the other or both so saying "All" would not always be correct.

Do you mean imported from external sources?  Those annotations will always be attached to all transcripts because we don't get a transcript ID from external sources.

**Comment by kimrutherford** [comment]
>  Those annotations will always be attached to all transcripts

Sorry, I was confused.  I've thought that through now and I understand what you mean.  I'll remove the "All".

**Comment by ValWood** [comment]
Yes sorry, that is where I first saw the annotation, I was looking at plasma membrane inferred annotations, I forgot how I ended up looking at the alternative transcripts

**Comment by ValWood** [comment]
I can't remember the decision on "All" but when I get back I will review alternative transcripts and their display to make sure it all makes sense. I think we only did a few test examples, but never fully annotated  all the transcripts. I also needed to review some things like tco1 where similar examples are annotated as ncRNAs, and make rules/document.

**Comment by ValWood** [comment]
Right I see, so the external sources were getting
SPAC17G6.02c.1
and
SPAC17G6.02c.2
instead of
SPAC17G6.02c (i.e. we don't know if it is one or both)

The current display was due to the filtering of one  annotation from the IBA and one from the ISS.

(Not this rule doesn't only apply to external sources, it applies anywhere we did not yet specify a specific transcript)

**Comment by kimrutherford** [comment]
> Right I see, so the external sources were getting
> SPAC17G6.02c.1
> and
> SPAC17G6.02c.2

Yep, that will always be the case because if we don't know the specific transcript we attach the GO annotation to all transcripts.  We don't have a way of attaching annotation to a gene directly.  That's true for GO annotation from any source, including Canto.

> The current display was due to the filtering of one annotation from the IBA and one from the ISS.

That sounds like a bug?  Could you make a separate issue for it?

**Comment by kimrutherford** [comment]
> I can't remember the decision on "All"

Here's the original issue: https://github.com/pombase/website/issues/1797#issuecomment-960554984

**Comment by ValWood** [comment]
I think this is a feature, not a bug. Will explain when I return. Not urgent.
